src/simplewallet/simplewallet.cpp

```diff
@@ -2656,23 +2699,37 @@ bool simple_wallet::ask_wallet_create_if_needed()
  * \brief Prints the seed with a nice message
  * \param seed seed to print
  */
-void simple_wallet::print_seed(std::string seed)
+void simple_wallet::print_seed(const epee::wipeable_string &seed)
 {
   success_msg_writer(true) << "\n" << tr("NOTE: the following 25 words can be used to recover access to your wallet. "
     "Write them down and store them somewhere safe and secure. Please do not store them in "
     "your email or on file storage services outside of your immediate control.\n");
-  boost::replace_nth(seed, " ", 15, "\n");
```

Comment:
Because wipeable_string does not have the iterator code that's needed for it to work.

Reply:
Add a mutable `char* wipeable_string::data()`, and use `epee::span` as indicated above.

Reply:
Since the code has to be changed to avoid leaving the string in some library buffer anyway, it's simple enough that I'll leave it as is.

Reply:
> Using span prevents the wipeable string size from beign modified by the call.
Can you expand on this ?

> std::cout and putchar will both have buffers in process memory anyway
Using the C API did not leave traces. Also that might be due to chance, though it was consistent (and the C++ API did also consistently leave many traces)

Reply:
>> Using span prevents the wipeable string size from beign modified by the call.
>
> Can you expand on this ?

If you pass `wipeable_string&` to a function, the size of the string can change. With `epee::span<char>` only the contents can change. I'm not even sure why I made that comment, because it is a fairly trivial point in this situation.

>> std::cout and putchar will both have buffers in process memory anyway
>
> Using the C API did not leave traces. Also that might be due to chance, though it was consistent (and the C++ API did also consistently leave many traces)

How did you analyze this? The C I/O functions are usually buffered too - [glibc `putchar`](https://github.com/bminor/glibc/blob/315304d24a5149a3670217514256b4ae959089e3/libio/bits/types/struct_FILE.h#L106) is **not** writing a character at a time to the operating system `write` function. There might be more traces with C++ somehow, although its not immediately obvious why that would be the case. I'd probably just use `boost::replace_nth` then do a C `puts`, pretty similar really. Meh I guess.

Reply:
I made a program to scan the memory used by the process and look for the data.